Bots/FAQNABot.cs

```diff
@@ -69,33 +70,20 @@ protected override async Task OnMembersAddedAsync(IList<ChannelAccount> membersA
         /// <param name="context">The current turn/execution flow.</param>
         /// <param name="text">Parses the text from user conversation.</param>
         /// <param name="cancellationToken">The cancellation Token.</param>
-        /// <returns>Approrpriate Card.</returns>
-        private async Task<List<Attachment>> GetMessageFromText(ITurnContext context, string text, CancellationToken cancellationToken)
+        /// <returns>Appropriate Card.</returns>
+        private async Task<List<Attachment>> GetCards(ITurnContext context, string text, CancellationToken cancellationToken)
         {
-            List<Attachment> nextMessage = null;
             var faqnaBotProvider = new FAQNABotProvider();
-            if (text == "hi"
-                     || text == "hello"
-                     || text == "reset"
-                     || text == "start over"
-                     || text == "restart")
-            {
-                // starts the conversation all over again from the welcome message,
-                // since the user has decided to restart the bot
-                nextMessage = new List<Attachment>() { await Task.Run(() => faqnaBotProvider.CreateWelcomeCardAttachment()) };
-            }
-            else if (context.Activity.Text == "Take a tour")
+
+            if (context.Activity.Text == "Take a tour")
             {
-                var carouselCards = new List<string>() { "carousel1", "carousel2", "carousel3" };
-                nextMessage = await Task.Run(() => faqnaBotProvider.CreateTourCardCarouselAttachment(carouselCards));
+                return await Task.Run(() => faqnaBotProvider.CreateTourCardCarouselAttachment());
             }
             else
             {
```

Comment:
@suramn Code is easy to follow and it does follow a pattern. The one suggestion that I would like to give you is that you can have a function/method which handles unrecognized input. I.e. SendUnrecognizedInput method that can then return an adaptive card which has a button that reads "Take a tour" on it.

Reply:
We are anyway sending the text user, that bot didn't understand the question and Welcome card with a  take a tour.
I can definitely add one, but just think in terms of simplicity.